### PR TITLE
[@types/react-highlight-words] Upgrading  typings to version 0.16.0

### PR DESCRIPTION
**Pull request by salguerooo**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/bvaughn/react-highlight-words/blob/master/src/Highlighter.js
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by kelly-tock — CHANGES_REQUESTED** [review]

File: types/react-highlight-words/index.d.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ interface HighlighterProps {
     /** The class name to be applied to an active match. Use along with activeIndex */
     activeClassName?: string;
     /** Specify the match index that should be actively highlighted. Use along with activeClassName */
-    activeIndex?: string;
+    activeIndex?: number;
```
[the remark this answers is not in the document]
Comment (salguerooo) [reply types/react-highlight-words/index.d.ts]: Thanks for the taking the time to review @kelly-tock!
I think the docs there need to be updated to reflect the actual implementation, which is this:
```
activeIndex: PropTypes.number,
```
(https://github.com/bvaughn/react-highlight-words/blob/master/src/Highlighter.js#L9)

I'll open a PR on the repo there to change the docs!

File: types/react-highlight-words/index.d.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ interface HighlighterProps {
      * Type of tag to wrap around highlighted matches; defaults to mark but can also be a React element
      * (class or functional)
      */
-    highlightTag?: string;
+    highlightTag?: string | React.ComponentType<any>;
```
[the remark this answers is not in the document]
Comment (salguerooo) [reply types/react-highlight-words/index.d.ts]: By looking at the implementation, it expects a component type (either a class or function), while ReactNode represents a rendered react element.

The place where it's used in the source code: https://github.com/bvaughn/react-highlight-words/blob/master/src/Highlighter.js#L121

File: types/react-highlight-words/index.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kelly-tock) [reply types/react-highlight-words/index.d.ts]: cool with me

File: types/react-highlight-words/index.d.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kelly-tock) [reply types/react-highlight-words/index.d.ts]: yep, you're right.

**Review by kelly-tock — APPROVED** [review]

File: types/react-highlight-words/index.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (salguerooo) [reply types/react-highlight-words/index.d.ts]: I made the PR and it got merged btw! No more confusion moving forward.

**Pull request merged**